### DISCUSSION TITLE
Global accessible data from hooks

### DISCUSSION THREAD

**Discussion by mhdaouas** [directus/directus#11811] [Archive]
Please I need to understand how to get some data from database and keep it in a const global variable to be able to access it from any hook? I need that because I have my workflow config in a specific collection and I need to access it to know what to do at each step of the workflow. I tried implementing the database read in different hooks like auth.login (I get denied access) and items.read (denied access) without success. Also seems like every extension auto reload deletes my global variable. What is the cleanest way to implement this? Appreciate your help :)

**Comment by rijkvanzanten** [comment]
What did you try so far? 

I would imagine you could create an async wrapper function, like:

```js
let config;

export async function getConfig() {
  if (!config) config = await getTheConfig();
  return config;
}
```

that way, whichever hook uses it first fetches it fresh, and every other invocation uses the cached existing variable 🙂
